### ISSUE TITLE
fopen failed to find the root CA certificate file error generated in mqtt_demo_mutual_auth example

### ISSUE THREAD

**Issue by pragnesh036** [aws/aws-iot-device-sdk-embedded-C#1724]
hello sir 
I am using stm32MP1 with your mqtt_demo_mutual_auth code and as wireless, I am using GSM with PPP.
If the network of GSM is good, then it will work fine, but in the bad network, my data sends perfectly After some back-off.
After two or three-day-long testings, I get errors according to bellow loges,

<details>
<summary>Log Output</summary>
<pre>
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

ppp0 is down
[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:269] SSL_connect failed to perform TLS handshake.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 65 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:314] fopen failed to find the root CA certificate file: ROOT_CA_PATH=/home/root/certificates/AmazonRootCA1.pem.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:628] Setting up credentials failed.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 936 ms backoff.

ppp0 is up
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 586 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 380 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 885 ms backoff.

 PPP0_NW  -->
 PPP0_switch  -->
Starting GSM
route: SIOCDELRT: No such process
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 4308 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 707 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 4629 ms backoff.

ppp0 is up
 PPP0_NW  -->
ppp0 is up
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 2039 ms backoff.

 PPP0_NW  -->
ppp0 is up
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 3161 ms backoff.

 PPP0_NW  -->
ppp0 is up
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 4535 ms backoff.

 PPP0_NW  -->
ppp0 is up
 PPP0_NW  -->
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 3034 ms backoff.

ppp0 is up
 PPP0_NW  -->
ppp0 is up
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 157 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 3276 ms backoff.

 PPP0_NW  -->
ppp0 is up
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 597 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 4288 ms backoff.

 PPP0_NW  -->
ppp0 is up
 PPP0_NW  -->
ppp0 is up
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 1609 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 725 ms backoff.

 PPP0_NW  -->
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 4291 ms backoff.

ppp0 is up
 PPP0_NW  -->
ppp0 is up
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 4491 ms backoff.

 PPP0_NW  -->
ppp0 is up
 PPP0_NW  -->
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 3484 ms backoff.

ppp0 is up
 PPP0_NW  -->
ppp0 is up
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 3739 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 74 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 803 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[WARN] [DEMO] [mqtt_demo_mutual_auth.c:843] Connection to the broker failed. Retrying connection after 414 ms backoff.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:821] Establishing a TLS session to a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com:8883.

[ERROR] [Sockets] [sockets_posix.c:135] Failed to resolve DNS: Hostname=a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com, ErrorCode=-11.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:674] Failed to establish a TLS connection.

[ERROR] [DEMO] [mqtt_demo_mutual_auth.c:838] Connection to the broker failed, all attempts exhausted.

[ERROR] [DEMO] [mqtt_demo_mutual_auth.c:2249] Failed to connect to MQTT broker a1386xddr5x240-ats.iot.ap-south-1.amazonaws.com. 

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:2268] Short delay before starting the next iteration....
</pre>
</details>

please any suggestion about this.

**Comment by dachalco** [comment]
@muneebahmed10 Could you please take a look at this?

**Comment by dachalco** [comment]
@pragnesh036

It's hard to say if something is wrong as this is a valid execution path for the the demo. It's attempted to connect using exponential backoff, fails, and continuously retries for a user configured number of times, before failing. 

Is your issue deterministically repeatable? Could it be that the device does not have sufficient signal to connect? Might be handy to periodically log GSM signal strength (RSSI) with timestamps, and juxtapose the reported signal strengths with the performance of your application.

**Comment by muneebahmed10** [comment]
Hi @pragnesh036,

I'm not sure why the `fopen` call fails. Some other process could have opened the file and `fopen` is unable to obtain the handle. However, as you can see from the logs, this only occurs once, and when the connection is retried the error occurs at different places, so it looks like your issue is different.

* Your first error `SSL_connect failed to perform TLS handshake` occurs in [`tlsHandshake`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/openssl_posix.c#L646) on line 646 of [Openssl_Connect](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/openssl_posix.c#L552).
* In the next try the error occurs earlier, the `fopen` error within [`setCredentials`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/openssl_posix.c#L617) on line 617 of Openssl_Connect
* Subsequent tries have an error earlier still, your [`Failed to resolve DNS`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/sockets_posix.c#L135) within [`resolveHostName`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/sockets_posix.c#L342) of [`Sockets_Connect`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/openssl_posix.c#L585) called on line 585 of Openssl_Connect.

Your main issue seems to be that you are unable to perform a DNS lookup. If you are missing or have a weak signal, then it would make sense that the DNS lookup would fail. You will need to do some investigation as to what is causing DNS to fail.

**Comment by pragnesh036** [comment]
> @pragnesh036
> 
> It's hard to say if something is wrong as this is a valid execution path for the demo. It's attempted to connect using exponential backoff, fails, and continuously retries for a user configured number of times, before failing.
> 
> Is your issue deterministically repeatable? Could it be that the device does not have sufficient signal to connect? Could be handy to periodically log GSM signal strength (RSSI) with timestamps, and juxtapose the reported signal strengths with your device's ability to connect.

Thank you for your reply.
We make a standalone device to monitor sensor parameters. 
At the field, our signal strength varies as per the location.
 the device will give not data on the proper timestamp It doesn't matter,
But in my case, if the device gives data at a different time, it is acceptable for me.
But after some time the code does not accept the path of the certificate. After this condition, the device doesn't send data on AWS.
The device only sends data when the code is rebooted.

**Comment by pragnesh036** [comment]
> Hi @pragnesh036,
> 
> I'm not sure why the `fopen` call fails. Some other process could have opened the file and `fopen` is unable to obtain the handle. However, as you can see from the logs, this only occurs once, and when the connection is retried the error occurs at different places, so it looks like your issue is different.
> 
> * Your first error `SSL_connect failed to perform TLS handshake` occurs in [`tlsHandshake`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/openssl_posix.c#L646) on line 646 of [Openssl_Connect](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/openssl_posix.c#L552).
> * In the next try the error occurs earlier, the `fopen` error within [`setCredentials`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/openssl_posix.c#L617) on line 617 of Openssl_Connect
> * Subsequent tries have an error earlier still, your [`Failed to resolve DNS`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/sockets_posix.c#L135) within [`resolveHostName`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/sockets_posix.c#L342) of [`Sockets_Connect`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/384da9dced0b70c8a7b3fcbf7e225f2b383739d7/platform/posix/transport/src/openssl_posix.c#L585) called on line 585 of Openssl_Connect.
> 
> Your main issue seems to be that you are unable to perform a DNS lookup. If you are missing or have a weak signal, then it would make sense that the DNS lookup would fail. You will need to do some investigation as to what is causing DNS to fail.

Our main issue is that after some long testing, the code does not accept the certificate path and after a reboot, the code device automatically accepts the path of the certificate.

**Comment by dan4thewin** [comment]
Hi, @pragnesh036 .  Is this a battery powered device?  I'm wondering if possibly some undervoltage situation is causing intermittent failures.

**Comment by pragnesh036** [comment]
> Hi, @pragnesh036 . Is this a battery-powered device? I'm wondering if possibly some Undervoltage situation is causing intermittent failures.

NO, our device is not a battery power device, we have an AC source that is converted into stable DC. If my GSM network is stable, then my issue will not generate, but I am preparing for the worst situation. Because in the working field the network is not predictable.

**Comment by yourslab** [comment]
@pragnesh036 So based on your log message, `fopen` is returning `NULL`, and there are many reasons why this can happen. To find this reason out, you can print `errno` by `#include <errno.h>` in `openssl_posix.c` and changing this [line of code](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/main/platform/posix/transport/src/openssl_posix.c#L317-L319) to
```
        LogError( ( "fopen failed to find the root CA certificate file: "
                    "ROOT_CA_PATH=%s, errno=%s.",
                    pRootCaPath, strerror( errno ) ) );
```
My guess is that some file descriptors are being leaked and hence you end up having to restart the machine in order to free those resources. With that being said, I will update the log message to make this easier for other customers to debug similar issues in the future.

**Comment by yourslab** [comment]
@pragnesh036 Any update on this issue?

**Comment by ravibhagavandas** [comment]
Where you able to get the logs displayed as @yourslab mentioned ?
From your logs, all the subsequent connections fails with DNS resolution error.  Error code -11 implies `Resource temporarily unavailable` for DNS resolution which indicates system is hitting some limit on resources. This could be due to some resource leaks as mentioned above. Did you make sure there are no open connections or file descriptors.

**Comment by muneebahmed10** [comment]
Hi @pragnesh036,

We will close this issue due to inactivity. If you have any other questions please open a new issue.
